### DISCUSSION TITLE
Can we update and remove URL entry that are generated by getStaticPaths.

### DISCUSSION THREAD

**Discussion by ats1999** [vercel/next.js#18078] [Help]
### I didn't found that this feature already exists in next js or not
When we use `getStaticPaths` then this method brings all the available URLs from the server and then renders the page using getStaticProps. This is great.

## Use case - update
Suppose i have a blog website when 1000s of blogs are posted on daily basis. A user `Rahul` posted a blog after few he noticed that blogs post need some changes. To do some changes he needs to rebuild the whole website. 

## use case - remove
After posting the blog, website owners noticed that this post has some harmful content and he wants to delete that post. 

In sort getStaticProps and getStaticPaths saves the URL in the JSON file, and i want to update the URL and remove the URL at run time.

**Comment by jamesmosier** [comment]
> Suppose i have a blog website when 1000s of blogs are posted on daily basis. A user Rahul posted a blog after few he noticed that blogs post need some changes. To do some changes he needs to rebuild the whole website.

You can use `getStaticProps` `revalidate` option for this use case: https://nextjs.org/docs/basic-features/data-fetching#getstaticprops-static-generation

With `revalidate` it will rebuild the content then serve that to the user once it is found to be "stale".

**Comment by ats1999** [comment]
As i know, revalidation will try to validate the data after an each interval. This what i don't want. I want to update the data only if it has been changed.

**Reply by mikiyasgirma** [reply]
https://www.youtube.com/watch?v=kdXKz1UWc3E&t=1262s
Forced Revalidation might be the solution for you
